### PR TITLE
cmd/containerboot: allow disabling secret storage in k8s.

### LINKED ISSUE #6704: FR:  TS_KUBE_SECRET cannot be unset. As it defaults to `tailscale` if it's `""`
### What are you trying to do?

Run ts in a kube environment, without the use of TS_KUBE_SECRET

### How should we solve this?

In the case that someone runs tailscale in a pod with other container(s), you can't give access to the secret to a single container on that pod, service account binds to the whole pod.

Please allow having empty TS_KUBE_SECRET, so people can set a TS_STATE_DIR of their liking and also pass TS_AUTH_KEY as they please too.

Thanks.

https://github.com/tailscale/tailscale/blob/7b65b7f85c6484cc852c201423ac577d9a4ceebf/cmd/containerboot/main.go#L40-L41

### What is the impact of not solving this?

When using it in a pod with other containers, you are force to bind a Service account to tall the containers inside that pod.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by danderson**
In some configurations, user explicitly do not want to store tailscale state in k8s secrets, because doing that leads to some annoying permission issues with sidecar containers. With this change, TS_KUBE_SECRET="" and TS_STATE_DIR=/foo will force storage to file when running in kubernetes.

Fixes #6704.

Signed-off-by: David Anderson <danderson@tailscale.com>

### CODE REVIEW

File: cmd/containerboot/main_test.go [hunk 1]
```diff
@@ -346,6 +346,30 @@ func TestContainerBoot(t *testing.T) {
 				},
 			},
 		},
+		{
+			Name: "kube_disk_storage",
+			Env: map[string]string{
+				"KUBERNETES_SERVICE_HOST":       kube.Host,
+				"KUBERNETES_SERVICE_PORT_HTTPS": kube.Port,
+				"TS_KUBE_SECRET":                "",
```
Comment (bradfitz) [inline cmd/containerboot/main_test.go:354]: add comment like `// testing that it's explicitly set to an empty value` ?

**Review by bradfitz — APPROVED** [review]

File: cmd/containerboot/main_test.go (hunk 1 above)
Comment (danderson) [reply cmd/containerboot/main_test.go]: Done.
